plugins/tweak/tweaks/material-size-for-melting.h

```diff
@@ -0,0 +1,121 @@
+#include <cmath>
+
+#include "modules/Materials.h"
+#include "modules/Random.h"
+
+#include "df/inorganic_raw.h"
+#include "df/item_armorst.h"
+#include "df/item_constructed.h"
+#include "df/item_glovesst.h"
+#include "df/item_helmst.h"
+#include "df/item_pantsst.h"
+#include "df/item_shieldst.h"
+#include "df/item_shoesst.h"
+#include "df/item_toolst.h"
+#include "df/item_trapcompst.h"
+#include "df/item_weaponst.h"
+
+static Random::MersenneRNG rng;
+static float get_random() {
+    return static_cast <float> (rng.drandom1());
+}
+
+static int32_t get_material_size_for_melting(df::item_constructed *item, int32_t base_material_size, float production_stack_size) {
+    const float melt_return_per_material_size = 0.3f, base_melt_recovery = 0.95f, loss_per_wear_level = 0.1f;
+
+    if (item->mat_type == 0) // INORGANIC only
+    {
+        float calculated_size;
+
+        auto inorganic = df::inorganic_raw::find(item->mat_index);
+        if (inorganic && inorganic->flags.is_set(df::inorganic_flags::DEEP_SPECIAL)){
+            calculated_size =static_cast<float>(base_material_size) / production_stack_size / melt_return_per_material_size;
+            // get size for melting, for adamantine forging cost == base_material_size, divided by amount of items created in batch
+        }
+        else {
+            calculated_size = std::max(std::floor(static_cast<float>(base_material_size) / 3.0f), 1.0f) / production_stack_size / melt_return_per_material_size;
+            // (std::max(std::floor(static_cast<float>(base_material_size) / 3.0f), 1.0f) / production_stack_size) - forging cost for non adamantine item
```

Comment:
While DFHack does not adhere to a strict "80 characters per line" rule, this is getting excessive.

Reply:
I have shortened it a bit.